src/Control/Email/Mailer.php

```diff
@@ -15,513 +13,124 @@ class Mailer extends Object
-            $headers["X-Priority"]  = 3;
-        }
-
-        // Merge!
-        $headers = array_merge($headers, $customHeaders);
-
-        // Headers 'Cc' and 'Bcc' need to have the correct case
-        foreach (array('Bcc', 'Cc') as $correctKey) {
-            foreach ($headers as $key => $value) {
-                if (strcmp($key, $correctKey) !== 0 && strcasecmp($key, $correctKey) === 0) {
-                    $headers[$correctKey] = $value;
-                    unset($headers[$key]);
-                }
-            }
-        }
-
-        return $headers;
-    }
-
-    /**
-     * Send a plain-text email.
-     *
-     * @param string $to Email recipient
-     * @param string $from Email from
-     * @param string $subject Subject text
-     * @param string $plainContent Plain text content
-     * @param array $attachedFiles List of attached files
-     * @param array $customHeaders List of custom headers
-     * @return mixed Return false if failure, or list of arguments if success
-     */
-    public function sendPlain($to, $from, $subject, $plainContent, $attachedFiles = array(), $customHeaders = array())
-    {
-        // Prepare plain text body
-        $fullBody = $this->encodeMessage($plainContent, $this->getMessageEncoding());
-        $headers["Content-Type"] = "text/plain; charset=utf-8";
-        $headers["Content-Transfer-Encoding"] = $this->getMessageEncoding();
-
-        // Send prepared message
-        return $this->sendPreparedMessage($to, $from, $subject, $attachedFiles, $customHeaders, $fullBody, $headers);
-    }
-
+    private $transport;
 
     /**
-     * Sends an email as a both HTML and plaintext
-     *
-     * @param string $to Email recipient
-     * @param string $from Email from
-     * @param string $subject Subject text
-     * @param string $htmlContent HTML Content
-     * @param array $attachedFiles List of attachments
-     * @param array $customHeaders User specified headers
-     * @param string $plainContent Plain text content. If omitted, will be generated from $htmlContent
-     * @return mixed Return false if failure, or list of arguments if success
+     * @var array Any recipients that didn't get the email
      */
-    public function sendHTML(
-        $to,
-        $from,
-        $subject,
-        $htmlContent,
-        $attachedFiles = array(),
-        $customHeaders = array(),
-        $plainContent = ''
-    ) {
-        // Prepare both Plain and HTML components and merge
-        $plainPart = $this->preparePlainSubmessage($plainContent, $htmlContent);
-        $htmlPart = $this->prepareHTMLSubmessage($htmlContent);
-        list($fullBody, $headers) = $this->encodeMultipart(
-            array($plainPart, $htmlPart),
-            "multipart/alternative"
-        );
-
-        // Send prepared message
-        return $this->sendPreparedMessage($to, $from, $subject, $attachedFiles, $customHeaders, $fullBody, $headers);
-    }
+    private $failedRecipients = array();
```

Comment:
I don't think that the service should be storing data related to specific invocations of Mailer::sendSwift(); I suggest to pass this back to Email::send() and have it stored on the email instance instead. Otherwise it dilutes the role of Mailer as a service.